src/nginx/t/auth_asymmetrickey.t

```diff
@@ -0,0 +1,311 @@
+$t->run_daemon(\&key, $t, $PubkeyPort, $rs256_pubkey, 'key.log');
+
+is($t->waitforsocket("127.0.0.1:${BackendPort}"), 1, 'Bookstore socket ready.');
+is($t->waitforsocket("127.0.0.1:${ServiceControlPort}"), 1, 'Service control socket ready.');
+is($t->waitforsocket("127.0.0.1:${PubkeyPort}"), 1, 'Pubkey socket ready.');
+
+$t->run();
+
+$response = ApiManager::http($NginxPort,<<"EOF");
+GET /shelves?key=this-is-an-api-key HTTP/1.0
+Host: localhost
+Authorization: Bearer $rs256_token
+
+EOF
+
+($response_headers, $response_body) = split /\r\n\r\n/, $response, 2;
+like($response_headers, qr/HTTP\/1\.1 200 OK/, 'Returned HTTP 200.');
+is($response_body, <<'EOF', 'Shelves returned in the response body.');
+{ "shelves": [
+    { "name": "shelves/1", "theme": "Fiction" },
+    { "name": "shelves/2", "theme": "Fantasy" }
+  ]
+}
+EOF
+
+$t->stop_daemons();
+$t->stop();
+################################################################################
+# RS256-signed jwt token is passed in query.
+
+$t->run_daemon(\&bookstore, $t, $BackendPort, 'bookstore.log');
+$t->run_daemon(\&servicecontrol, $t, $ServiceControlPort, 'servicecontrol.log');
+$t->run_daemon(\&key, $t, $PubkeyPort, $rs256_pubkey, 'key.log');
+
+is($t->waitforsocket("127.0.0.1:${BackendPort}"), 1, 'Bookstore socket ready.');
+is($t->waitforsocket("127.0.0.1:${ServiceControlPort}"), 1, 'Service control socket ready.');
+is($t->waitforsocket("127.0.0.1:${PubkeyPort}"), 1, 'Pubkey socket ready.');
+
+$t->run();
+
+$response = ApiManager::http($NginxPort,<<"EOF");
+GET /shelves?key=this-is-an-api-key&access_token=$rs256_token HTTP/1.0
+Host: localhost
+
+EOF
+
+($response_headers, $response_body) = split /\r\n\r\n/, $response, 2;
+like($response_headers, qr/HTTP\/1\.1 200 OK/, 'Returned HTTP 200.');
+is($response_body, <<'EOF', 'Shelves returned in the response body.');
+{ "shelves": [
+    { "name": "shelves/1", "theme": "Fiction" },
+    { "name": "shelves/2", "theme": "Fantasy" }
+  ]
+}
+EOF
+
+$t->stop_daemons();
+$t->stop();
+################################################################################
+
+sub bookstore {
+  my ($t, $port, $file) = @_;
+  my $server = HttpServer->new($port, $t->testdir() . '/' . $file)
+    or die "Can't create test server socket: $!\n";
+  local $SIG{PIPE} = 'IGNORE';
+
+  $server->on('GET', '/shelves', <<'EOF');
+HTTP/1.1 200 OK
+Connection: close
+
+{ "shelves": [
+    { "name": "shelves/1", "theme": "Fiction" },
+    { "name": "shelves/2", "theme": "Fantasy" }
+  ]
+}
+EOF
+
+  $server->run();
+}
+
+sub servicecontrol {
+  my ($t, $port, $file) = @_;
+  my $server = HttpServer->new($port, $t->testdir() . '/' . $file)
+    or die "Can't create test server socket: $!\n";
+  local $SIG{PIPE} = 'IGNORE';
+
+  $server->on('POST', '/v1/services/endpoints-test.cloudendpointsapis.com:check', <<'EOF');
+HTTP/1.1 200 OK
+Connection: close
+
+EOF
+
+  $server->run();
+}
+
+sub key {
+  my ($t, $port, $secret, $file) = @_;
```

Comment:
can we pass all secrets here, and use one at time? so not to stop and restart daemon

Reply:
Done.